front/lib/resources/internal_mcp_server_in_memory_resource.ts

```diff
@@ -30,6 +30,43 @@ import { removeNulls } from "@app/types";
 
 const METADATA_CACHE_TTL_MS = 5 * 60 * 1000; // 5 minutes
 
+// Getting the metadata is a relatively long operation, so we cache it for 5 minutes
```

Comment:
What makes it long given that it's all in memory here? How many DB call do we have behind this ooc?